wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt

```diff
@@ -965,38 +950,165 @@ class SwiftGenerator private constructor(
           .build(),
       )
     }
-    type.oneOfs.forEach { oneOf ->
-      val enumName = oneOfEnumNames.getValue(oneOf).makeOptional()
-      addParameter(
-        ParameterSpec.builder(oneOf.name, enumName)
-          .defaultValue("nil")
+    if (includeOneOfs) {
+      type.oneOfs.forEach { oneOf ->
+        val enumName = oneOfEnumNames.getValue(oneOf).makeOptional()
+        addParameter(
+          ParameterSpec.builder(oneOf.name, enumName)
+            .defaultValue("nil")
+            .build(),
+        )
+      }
+    }
+  }
+
+  private fun TypeSpec.Builder.generateMessageStorageDelegateConstructor(
+    type: MessageType,
+    storageName: String,
+    structType: DeclaredTypeName,
+    storageType: DeclaredTypeName,
+    oneOfEnumNames: Map<OneOf, DeclaredTypeName>,
+    fileMembers: MutableList<FileMemberSpec>,
+  ) {
+    addFunction(
+      FunctionSpec.constructorBuilder()
+        .addModifiers(PUBLIC)
+        .addParameters(
+          type,
+          oneOfEnumNames,
+          includeDefaults = false,
+          includeOneOfs = false,
+          fieldsFilter = { it.isRequiredParameter },
+        )
+        .addParameter(
+          ParameterSpec.builder(
+            "configure",
+            FunctionTypeName.get(
+              TypeVariableName.typeVariable("inout Self.${storageType.simpleName}"),
+              returnType = VOID,
+            ),
+          )
+            .defaultValue("{ _ in }")
+            .build(),
+        )
+        .apply {
+          val storageParams = mutableListOf<CodeBlock>()
+          type.fields.filter { it.isRequiredParameter }.forEach { field ->
+            storageParams += CodeBlock.of("%1N: %1N", field.name)
+          }
+          storageParams += CodeBlock.of("%1N: %1N", "configure")
+          addStatement(
+            "self.%N = %T(\n%L\n)",
+            storageName,
+            storageType,
+            storageParams.joinToCode(separator = ",\n"),
+          )
+        }
+        .build(),
+    )
+
+    if (type.fields.all { it.isRequiredParameter } && type.oneOfs.isEmpty()) {
+      return
+    }
+
+    val memberwiseExtension = ExtensionSpec.builder(structType)
```

Comment:
Maybe add a comment noting we want to remove this, but it's left in temporarily for compatibility.

Reply:
- [ ] Add deprecation annotations